Menu/PageTreeMenuBuilder.php

```diff
@@ -72,6 +72,12 @@ private function appendNodes (ItemInterface $parent, array $nodes, array $option
     {
         foreach ($nodes as $node)
         {
+            // Don't render nodes that the User is not supposed/eligible to see
```

Comment:
(and we need to add a note about other template implementations needing to duplicate this logic)